src/main/java/core/basesyntax/temp/FigureSupplier.java

```diff
@@ -0,0 +1,44 @@
+package core.basesyntax.temp;
+
+import core.basesyntax.figures.Circle;
+import core.basesyntax.figures.IsoscelesTrapezoid;
+import core.basesyntax.figures.Rectangle;
+import core.basesyntax.figures.RightTriangle;
+import core.basesyntax.figures.Square;
+import java.util.Random;
+
+public class FigureSupplier {
+    private Figure rightTriangle = new RightTriangle();
+    private Figure isoscelesTrapezoid = new IsoscelesTrapezoid();
+    private Figure square = new Square();
+    private Figure circle = new Circle();
+    private Figure rectangle = new Rectangle();
+    private Figure[] figure = new Figure[]{rightTriangle,
+            isoscelesTrapezoid, square, circle, rectangle};
+
+    public Figure getRandomFigure() {
+        int index = new Random().nextInt(figure.length);
+        Figure randomFigure = figure[index];
+        randomFigure.setColor(new ColorSupplier().getRandomColor());
```

Comment:
same, let's make Random and ColorSupplier as class level variables, not to create new instances every time we call getRandomFigure() method.